libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewApi.kt

```diff
@@ -1,108 +1,109 @@
-     * between the first and the last point.
+     * Sets a [point] as destination and request [NavigationView] to enter Destination Preview state.
      *
-     * @param points list of [Point]
+     * [NavigationViewListener.onDestinationChanged] will be called once new destination is set
+     * and [NavigationViewListener.onDestinationPreview] once [NavigationView] enters
+     * Destination Preview state.
      */
-    fun fetchRoutes(points: List<Point>) {
-        store.dispatch(RoutePreviewAction.FetchPoints(points))
-    }
+    abstract fun startDestinationPreview(point: Point)
 
     /**
-     * Sets the [routes] passed to [NavigationView] and not to [MapboxNavigation]. The action
-     * does not triggers [RouteProgress], [BannerInstructions] or [VoiceInstructions] updates even
-     * if the [TripSessionState] is [TripSessionState.STARTED].
+     * Request [NavigationView] to enter Route Preview state.
+     *
+     * [NavigationViewListener.onRoutePreview] will be called once [NavigationView] enters
+     * Route Preview state.
      *
-     * @param routes list of [NavigationRoute]
+     * Fails with an error when either Destination or Preview Routes has not been set.
      */
-    fun setPreviewRoutes(routes: List<NavigationRoute>) {
-        store.dispatch(RoutePreviewAction.Ready(routes))
-    }
+    abstract fun startRoutePreview(): Expected<Throwable, Unit>
 
     /**
-     * Sets the [routes] passed to [MapboxNavigation]. The action triggers [RouteProgress],
-     * [BannerInstructions] and [VoiceInstructions] updates if the [TripSessionState] is
-     * [TripSessionState.STARTED].
+     * Sets a preview [routes] and request Request [NavigationView] to enter Route Preview state.
+     * Last [DirectionsWaypoint] location will be used as the destination.
+     *
+     * [NavigationViewListener.onDestinationChanged] will be called once new destination is set
+     * and [NavigationViewListener.onRoutePreview] will be called once [NavigationView] enters
+     * Route Preview state.
      *
-     * @param routes list of [NavigationRoute]
+     * Fails with an error when [routes] is an empty list.
      */
-    fun setRoutes(routes: List<NavigationRoute>) {
-        store.dispatch(RoutesAction.SetRoutes(routes))
-    }
+    abstract fun startRoutePreview(routes: List<NavigationRoute>): Expected<Throwable, Unit>
 
     /**
-     * Sets a destination to [NavigationView].
-     * @property point the destination location
+     * Request [NavigationView] to enter Active Navigation state.
+     *
+     * [NavigationViewListener.onActiveNavigation] will be called once [NavigationView] enters
+     * Route Preview state.
+     *
+     * Fails with an error when either Destination or Preview Routes has not been set.
      */
-    fun setDestination(point: Point) {
-        store.dispatch(DestinationAction.SetDestination(Destination(point)))
-    }
+    abstract fun startNavigation(): Expected<Throwable, Unit>
```

Comment:
```suggestion
    abstract fun startActiveGuidance(): Expected<Throwable, Unit>
```
to be consistent with our `Free Drive` / `Active Guidance` terminology